### ISSUE TITLE
Video is shaded

### ISSUE THREAD

**Issue by mairin** [vector-im/riot-android#885]
I'm using riot.im on Android v 0.6.5 (ebddfd4) on a Samsung Galaxy S5. When I use it to video chat, my image always looks very dark / shaded, like some kind of filter was put on top of it. You can only see my very faint silhouette and lights and nothing else. If I switch to the front camera, the image is fine. Verified with a friend this is the case both on my end and on their end.

**Comment by mairin** [comment]
Might be this? http://stackoverflow.com/questions/20609011/front-camera-take-very-dark-capture-in-android

**Comment by ylecollen** [comment]
Is it an official ROM ?
The android client does not setup the camera. It is done by a third party lib.

**Comment by ylecollen** [comment]
We don't manage the camera initialisation.
It works properly on a Samsung s3, s4 and S6.
I will try to find a samsung S5.

**Comment by mairin** [comment]
Its a rooted official ROM. Its android 6.0.1 sw version g900pvps3cpe2

**Comment by ylecollen** [comment]
Do you have the same behaviour with any other application which use the front camera ?

**Comment by ylecollen** [comment]
![screenshot_20170125-085656](https://cloud.githubusercontent.com/assets/8971031/22282693/61fde3ea-e2de-11e6-90c4-90d554441013.png)

**Comment by ylecollen** [comment]
I succeeded to reproduce your issue on some devices (Samsung S6 or a ZE550KL).

To fix it, you should just need to move your device and yourself of course :). 
Could you check if it works if you move your device ?

It is linked to the exposure compensation. I could update the default value but i'm not it will work on any device.

**Comment by mairin** [comment]
No, it doesnt happen on any other app. I tried android camera app, facebook messenger, twitter, google hangouts and couldnt reproduce.

On January 25, 2017 2:03:50 AM EST, Yannick LE COLLEN <notifications@github.com> wrote:
>Do you have the same behaviour with any other application which use the
>front camera ?
>
>-- 
>You are receiving this because you were assigned.
>Reply to this email directly or view it on GitHub:
>https://github.com/vector-im/riot-android/issues/885#issuecomment-275034505

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by mairin** [comment]
So I tried it while standing outside and it worked ok. I had problems when i filed the bug and i was indoors and it was night (but i was in a well lit room.) I think when i initially tried it, phone was face up on dining room table under a bright chandelier and subsequent calls done in different rooms with normal indoor lighying conditions continued to be blown out. Are you setting exp compensation based on first call attempt and saving rhat value? It seemed like i couldnt clear it from what ot was set to while facing chandelier... i tried reboots too. 

On January 25, 2017 3:23:11 AM EST, Yannick LE COLLEN <notifications@github.com> wrote:
>I succeeded to reproduce your issue on some devices (Samsung S6 or a
>ZE550KL).
>
>To fix it, you should just need to move your device and yourself of
>course :). 
>Could you check if it works if you move your device ?
>
>It is linked to the exposure compensation. I could update the default
>value but i'm not it will work on any device.
>
>
> 
>
>
>-- 
>You are receiving this because you were assigned.
>Reply to this email directly or view it on GitHub:
>https://github.com/vector-im/riot-android/issues/885#issuecomment-275046567

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by ylecollen** [comment]
Did you call with hangout or did you only try to take a photo ?

 I have same issue while VOIPing with Hangout : i.e my own video is very dark.

 The application does not play with the camera settings (it is not obvious to find the right settings to each device).
 Most of the application uses the device camera application.

**Comment by ShahabJafri** [comment]
I have the same rooted Galaxy S5 G900F and I am facing the same issue as @mairin . I had reported it twice from the Riot app itself attaching the screenshots. No other software has this problem: I've tested Wire, Skype, Hangouts. 
![screenshot_2017-01-30-22-31-47](https://cloud.githubusercontent.com/assets/5832511/22554976/31904158-e973-11e6-8258-4a60cca4b2fe.jpg)

I did try to move the device and no joy.

**Comment by Sorunome** [comment]
I experience the same issue with an ALCATEL Pixi 4 (Android 6.0). The front camera is just extremely dark (like in the picture above), while other apps are able to handle the lighting just fine

**Comment by xuhdev** [comment]
I can confirm the issue reported by the OP on Galaxy S5.

**Comment by lieuwex** [comment]
FWIW: Having this issue too on a Nexus 6 while other apps that use the camera in video mode too don't have this problem.
